src/pkcs11/framework-pkcs15.c

```diff
@@ -5276,13 +5354,41 @@ static CK_RV
 get_ec_pubkey_params(struct sc_pkcs15_pubkey *key, CK_ATTRIBUTE_PTR attr)
 {
 	struct sc_ec_parameters *ecp;
+	unsigned long expected_size = 0;
+	unsigned char *out = NULL;
 
 	if (key == NULL)
 		return CKR_ATTRIBUTE_TYPE_INVALID;
 	if (key->alg_id == NULL)
 		return CKR_ATTRIBUTE_TYPE_INVALID;
 
 	switch (key->algorithm) {
+	case SC_ALGORITHM_EDDSA:
+#define CURVE "edwards25519"
+		/* TODO key->alg_id->oid contains OID which we need to convert to curve name */
```

Comment:
This would require having the table of supported ED curves also in the pkcs11 layer to match and convert them (we have already one in pkcs15 emulation and openpgp driver) or to have the curve name converted in some of the structure members. I wanted to keep it "simple" as we support only one now.

Reply:
FYI in the end, I changed encoding of EC_PARAMS to OID and implemented also this part during rebasing to current master so this can be considered resolved.

Reply:
if you added `edwards25519` to `ec_curve_infos[]`  in `pkcs15-pubkey.c `sc_pkcs15_fix_ec_parameters` could be used to address `/* TODO key->alg_id->oid contains OID which we need to convert to curve name */`

Reply:
Right. It is already there so this chunk is outdated and conversion to the EC_PARAMS is already done.